src/Combinatorics/Matroids/matroid_strata_grassmannian.jl

```diff
@@ -0,0 +1,378 @@
+        
+        col_b = setdiff(b,B)[1]
+        col_b = col_b - length([a for a in B if a ≤ col_b]) 
+        
+        push!(new_coords, [row_b,col_b])
+    end
+    return sort!(new_coords, by = x -> (x[2], x[1]))    
+end
+
+
+
+# Given the bases Bs of a matroid, a fixed basis B, and a coefficient field F
+# this function creates a polynomial ring in xij, where the xij are 
+# determined by the function basis_matrix_coordinates. This function also
+# returns (as the 3rd element of a triple) a dictionary (i,j) => xij.
+
+function make_polynomial_ring(Bs::Vector{Vector{Int}}, B::Vector{Int},
+                              F::AbstractAlgebra.Ring)
+    
+    MC = bases_matrix_coordinates(Bs, B)
+    R, x = PolynomialRing(F, :"x"=>MC)
+    xdict = Dict{Vector{Int}, MPolyElem}([MC[i] => x[i] for i in 1:length(MC)])
+    return R, x, xdict
+end
+
+
+# This function returns a d x (n-d) matrix with values in the polynomial ring
+# created by make_polynomial_ring. The entries are xij, except where the
+# value is 0, as determined by the nonbases. 
+
+function make_coordinate_matrix_no_identity(d::Int, n::Int,
+                                            MC::Vector{Vector{Int}},
+                                            R::MPolyRing, x::Vector{T},
+                                            xdict::Dict{Vector{Int}, MPolyElem}) where T <: MPolyElem
+    
+    S = MatrixSpace(R, d, n-d)
+    X = S()
+    for j in 1:n-d, i in 1:d
+        if [i,j] in MC
+            X[i,j] = xdict[[i,j]]
+        else
+            X[i,j] = R(0)
+        end
+    end
+    return X
+end
+
+
+# M and N have same number of rows, and M has #B columns, both have entries in ring R
+function interlace_columns(M, N, B::Vector{Int},
+                           R::MPolyRing, x::Vector{T}) where T <: MPolyElem 
+    
+    M_nrows, M_ncols = size(M)
+    N_nrows, N_ncols = size(N)
+    n = M_ncols + N_ncols
+
+    S = MatrixSpace(R, M_nrows, n)
+    Bc = [i for i in 1:n if !(i in B)]
+    
+    X = S()
+    X[:, B] = M
+    X[:, Bc] = N
+    
+    return X 
+end
+
+# This makes the matrix X from which we compute the coordinate ring of the matroid
+# stratum. It has the identity matrix at columns indexed by B, 0's at locations
+# determined by the nonbases of X. 
+function make_coordinate_matrix(d::Int, n::Int, MC::Vector{Vector{Int}},
+                                B::Vector{Int},
+                                R::MPolyRing, x::Vector{T},
+                                xdict::Dict{Vector{Int}, MPolyElem}) where T <: MPolyElem
+    
+    Id = identity_matrix(R,d)
+    Xpre = make_coordinate_matrix_no_identity(d, n, MC, R, x, xdict)
+    return interlace_columns(Id, Xpre, B, R, x)
+end
+
+
+# This function returns all d x d determinants of the matrix X from above
+# of all collections of d-columns coming from the bases of the matroid.
+
+function bases_determinants(X::Matrix{T}, Bs::Vector{Vector{Int}})  where {T<:MPolyElem}
```

Comment:
```suggestion
function bases_determinants(X::MatrixElem{T}, Bs::Vector{Vector{Int}})  where {T<:MPolyElem}
```
This looked like a julia matrix to me. Probably, Oscar matrices are, what you really want here?

Reply:
I implemented this suggestion, now it works.